### DISCUSSION TITLE
Is it possible to select used DSO catalogs in script

### DISCUSSION THREAD

**Discussion by b-matous** [Stellarium/stellarium#2454] [Q&A] [answered]
I wonder if there is a possibility to choose what catalogs of DSO are visible within the script.
E. g. I will have a script which will show only NGC objects no matter what the user has set.
In the help of NebulaMgr, there is a function setCatalogFilters which looks like what I want but I fail to use it, since I have to feed it with CatalogGroup object which I do not know how to construct.

**Comment by alex-w** [comment, marked as answer]
Please see enum CatalogGroupFlags: https://github.com/Stellarium/stellarium/blob/56c1d54f6e15946c69b6c5d333632562b9ddc967/src/core/modules/Nebula.hpp#L44

**Reply by b-matous** [reply]
That is good.
But I do not know how to make it work in the script.
If I put the value as integer I get error:
```
script error: "TypeError: cannot call setCatalogFilters(): argument 1 has unknown type `Nebula::CatalogGroup' (register the type with qScriptRegisterMetaType())"
```
If I try to construct it with type name I also fail
`NebulaMgr.setCatalogFilters(Nebula::CatalogGroup(1));`
`script error: "SyntaxError: Parse error" @ line 1`

**Reply by gzotti** [reply]
Apparently this function was so far never used in scripts. Given that QtScript is deprecated and must be removed soon, it seems not useful to add such a function now. I hope the QtScript replacement that we need to develop can then be extended to handle your case. Sorry for the inconvenience.
